src/utils/mdast-to-vdom.js

```diff
@@ -60,6 +61,9 @@ class VDOMTransformer {
       this._handlers[type] = (cb, node, parent) => VDOMTransformer.handle(cb, node, parent, that);
       return true;
     });
+
+    this._headingHandler = new HeadingHandler(options);
+    this.match('heading', this._headingHandler.handler());
```

Comment:
As the heading handler needs to support resetting the "counters" for the headings, we need a slightly more complex construct for it